src/test/concurrency/stress_test.cpp

```diff
@@ -0,0 +1,147 @@
+    Hyrise::get().set_scheduler(std::make_shared<NodeQueueScheduler>());
+  }
+};
+
+TEST_F(StressTest, TestTransactionConflicts) {
+  // Update a table with two entries and a chunk size of 2. This will lead to a high number of transaction conflicts
+  // and many chunks being created
+  auto initial_sum = int64_t{};
+  {
+    auto pipeline = SQLPipelineBuilder{std::string{"SELECT SUM(a) FROM table_a"}}.create_pipeline();
+    const auto [_, table] = pipeline.get_result_table();
+    initial_sum = table->get_value<int64_t>(ColumnID{0}, 0);
+  }
+
+  const std::string sql = "UPDATE table_a SET a = a + 1 WHERE a = (SELECT MIN(a) FROM table_a);";
+
+  std::atomic_int successful_increments{0};
+  std::atomic_int conflicted_increments{0};
+  const auto iterations_per_thread = 20;
+
+  const auto run = [&]() {
+    int my_successful_increments{0};
+    int my_conflicted_increments{0};
+    for (auto iteration = 0; iteration < iterations_per_thread; ++iteration) {
+      auto pipeline = SQLPipelineBuilder{sql}.create_pipeline();
+      const auto [status, _] = pipeline.get_result_table();
+      if (status == SQLPipelineStatus::Success) {
+        ++my_successful_increments;
+      } else {
+        ++my_conflicted_increments;
+      }
+    }
+    successful_increments += my_successful_increments;
+    conflicted_increments += my_conflicted_increments;
+  };
+
+  const auto num_threads = 100u;
+  std::vector<std::future<void>> thread_futures;
+  thread_futures.reserve(num_threads);
+
+  for (auto thread_num = 0u; thread_num < num_threads; ++thread_num) {
+    // We want a future to the thread running, so we can kill it after a future.wait(timeout) or the test would freeze
+    thread_futures.emplace_back(std::async(std::launch::async, run));
+  }
+
+  for (auto& thread_future : thread_futures) {
+    // We give this a lot of time, not because we need that long for 100 threads to finish, but because sanitizers and
+    // other tools like valgrind sometimes bring a high overhead that exceeds 10 seconds.
+    if (thread_future.wait_for(std::chrono::seconds(150)) == std::future_status::timeout) {
+      ASSERT_TRUE(false) << "At least one thread got stuck and did not commit.";
+      thread_future.get();
+    }
+  }
+
+  auto final_sum = int64_t{};
+  {
+    auto pipeline = SQLPipelineBuilder{std::string{"SELECT SUM(a) FROM table_a"}}.create_pipeline();
+    const auto [_, table] = pipeline.get_result_table();
+    final_sum = table->get_value<int64_t>(ColumnID{0}, 0);
+  }
+
+  // Really pessimistic, but at least 2 statements should have made it
+  EXPECT_GT(successful_increments, 2);
+
+  EXPECT_EQ(successful_increments + conflicted_increments, num_threads * iterations_per_thread);
+  EXPECT_FLOAT_EQ(final_sum - initial_sum, successful_increments);
+}
+
+TEST_F(StressTest, TestTransactionInserts) {
+  // An insert-heavy load on a table with a ridiculously low max chunk size, creating many new chunks
```

Comment:
added "This is different from TestTransactionConflicts, which uses updates, in that each iteration is guaranteed to create a row. In the other test, a failed "mark for deletion" (i.e., swap of the row's tid) would lead to no row being appended."